src/expression-helpers.js

```diff
@@ -37,6 +41,7 @@ function getObservableValue_fromDynamicKey_fromObservable(key, root, helperOptio
 	}, function setDynamicKey(newVal){
 		stacheKey.write(canReflect.getValue(root), getKeys(), newVal);
 	});
+	// This prevents lazy evalutaion
```

Comment:
The `stacheKey.read` above is still going to cause dynamic keys to be read... something like this fails:

```
	view = stache("<div>{{#and(this.isFalse, [ this.key ])}}TRUE{{else}}FALSE{{/and}}</div>");

	fragment = view({
		isFalse: false,
		key: "shouldNotBeRead",
		shouldNotBeRead: new Observation(function avoidReadingThis(){
			QUnit.ok(false, "should not be read");
		})
	});
```

Reply:
Oh, I realize now that's what the comment above means.